src/components/productrecommendation.fetchallpages.jsx

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright © 2018 Elastic Path Software Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { login } from '../utils/AuthService';
+import ProductListItemMain from './productlistitem.main';
+
+const Config = require('Config');
+
+
+class Productrecommendationfetchallpages extends React.Component {
+  static propTypes = {
+    product: PropTypes.objectOf(PropTypes.any).isRequired,
+    length: PropTypes.number.isRequired,
+    count: PropTypes.number.isRequired,
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      links: [],
+    };
+  }
+
+  componentDidMount() {
+    const { product } = this.props;
+    const { links } = this.state;
+    const link = links;
+    const arr = product.links;
+    for (let i = 0; i < arr.length; i += 1) {
+      if (arr[i].rel === 'element') {
+        link.push(arr[i]);
+      } else if (arr[i].rel === 'next') {
+        this.renderUrl(arr[i].uri);
+      }
+    }
+    this.setState({
+      links: link,
+    });
+  }
+
+  renderUrl(uri) {
```

Comment:
you shouldn't call this renderUrl, you're not doing anything related to rendering here. consider renaming to something along the lines of expandurl.
also, it would make sense to be consistent with uri/url. your function name says url, but it takes uri.

Reply:
expandUrl it is ! :)